nilearn/decoding/decoder.py

```diff
@@ -666,6 +745,35 @@ def _binarize_y(self, y):
         self.n_classes_ = len(self.classes_)
         return y
 
+    def _predict_dummy(self, n_samples):
+        """Non-sparse scikit-learn based prediction steps for classification
+           and regression"""
+        dummy_output = self.dummy_output_
+
+        if isinstance(self.estimator, DummyClassifier):
+            estimator_params = self.estimator.get_params()
+            strategy = estimator_params['strategy']
+            if strategy in ['most_frequent', 'prior']:
+                scores = np.tile([dummy_output[k].argmax() for
+                                  k in range(self.n_outputs_)], [n_samples, 1])
+            elif strategy == 'stratified':
+                rs = check_random_state(0)
+                proba = []
+                for k in range(self.n_outputs_):
+                    out = rs.multinomial(1, dummy_output[k], size=n_samples)
+                    out = out.astype(np.float64)
+                    proba.append(out)
+                scores = np.vstack([proba[k].argmax(axis=1) for
+                                    k in range(self.n_outputs_)]).T
+            else:
+                message = ('Implemented only for strategies "most_frequent",'
```

Comment:
One detail  : Codecov reports that this paragraph is not tested, whereas there is a test on this edgecase. But this seems to be a duplicate of the test at new line 119. If you feel both are useful please test this one as well, otherwise I guess the current test 769  would be sufficient since it would be executed independently of the API level used so the other could be removed I guess

Reply:
Good catch.